ui/hooks/gasFeeInput/useGasFeeInputs.js

```diff
@@ -168,6 +172,20 @@ export function useGasFeeInputs(
         setEstimateUsed(transaction?.userFeeLevel);
         setInternalEstimateToUse(transaction?.userFeeLevel);
       }
+
+      const maximumGas = new Numeric(transaction?.txParams?.gas ?? '0x0', 16)
+        .times(new Numeric(transaction?.txParams?.maxFeePerGas ?? '0x0', 16))
+        .toPrefixedHexString();
+
+      const fee = new Numeric(maximumGas, 16, EtherDenomination.WEI)
+        .toDenomination(EtherDenomination.ETH)
+        .toBase(10)
+        .toString();
+
+      if (Number(fee) > Number(GAS_LIMIT_TOO_HIGH_IN_ETH)) {
+        setEstimateUsed(PriorityLevels.dappSuggestedHigh);
+      }
```

Comment:
maybe we can reorganize to avoid setting the estimate twice since setEstimateUsed may be called above if `transaction?.userFeeLevel`